### PR TITLE
Fix fn level secrets

### PR DESCRIPTION
**Pull request by sneha-nagole**
##What is changed
1. Common Function level secrets are resolved correctly without being overridden by the last function level secret.
2. Serverless resolves variables after plugins are loaded. This is causing an issue when function configuration uses variables (file imports or simple variables). As a fix, re-initializing configuration after Serverless resolves the variables in Serverless config file.

We can improve the solution by splitting config.options & config.environments into separate variables. I am planning to make that change when we propose a PR on the trek10inc serverless-secrets repo.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,14 +1,14 @@
 {
   "name": "serverless-secrets",
-  "version": "3.0.0-beta.11",
+  "version": "3.0.0-beta.12",
```
Comment (comeara) [inline package.json]: I don't think it is a good idea to change the package information. Our end goal is to get this integrated back into the main fork. The original author will have to revert these changes in order to release a new version.

File: plugin/index.js [hunk 2]
```diff
@@ -215,46 +216,53 @@ class ServerlessSecrets {
   }
 
   generateConfig () {
-    this.serverless.cli.log('Generating Serverless Secrets Config')
-    if (!this.serverless.service.provider.name) {
-      throw new Error('No provider name configured in serverless.yml')
-    }
+      this.serverless.cli.log('Generating Serverless Secrets Config options')
+      if (!this.serverless.service.provider.name) {
```
Comment (comeara) [inline plugin/index.js]: This looks like changes to whitespace. The PR back to the original fork will be clearer without whitespace changes.

Comment (sneha-nagole) [reply plugin/index.js]: Thank you for the feedback. Changes are made & ready for review again.

File: package.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
   },
   "repository": {
     "type": "git",
-    "url": "https://github.com/trek10inc/serverless-secrets"
```
Comment (comeara) [inline package.json]: I recommend reverting this for a cleaner PR going to the upstream repo.

File: package.json [hunk 4]
```diff
@@ -21,12 +21,12 @@
     "parameter store",
     "env"
   ],
-  "author": "Jared Short <jshort@trek10.com> (https://www.trek10.com)",
```
Comment (comeara) [inline package.json]: I recommend reverting this for a cleaner PR going to the upstream repo.

File: package.json [hunk 5]
```diff
@@ -21,12 +21,12 @@
...
+  "author": "Sneha Nagole <sneha.nagole@tabordasolutions.com>",
   "license": "MIT",
   "bugs": {
     "url": "https://github.com/trek10inc/serverless-secrets/issues"
   },
-  "homepage": "https://github.com/trek10inc/serverless-secrets",
```
Comment (comeara) [inline package.json]: I recommend reverting this for a cleaner PR going to the upstream repo.

File: package.json (hunk 4 above)
Comment (sneha-nagole) [reply package.json]: Modified source code based on the review.

File: package.json (hunk 5 above)
Comment (sneha-nagole) [reply package.json]: Modified source code based on the review.

File: package.json (hunk 3 above)
Comment (sneha-nagole) [reply package.json]: Modified source code based on the review.

**Review by comeara — APPROVED** [review]
👍
